Drop skipperOauthOidc cookies in oauthOidc filter

AndreasSko opened the pull request:
The `skipperOauthOidc` cookies are only intended for Skipper, so they don't need to be forwarded to the application.

With this PR we simply drop those cookies before forwarding the request. This will reduce the overall header size of requests.
The implementation is inspired by the [`oauthGrant` filter](https://github.com/zalando/skipper/blob/6b448b1fe90cc113e365be8fba7cd6d122ad7a6d/filters/auth/grantcookie.go#L93-L100)

Closes https://github.com/zalando/skipper/issues/3459

AlexanderYastrebov commented on filters/auth/oidc_test.go:
```diff
@@ -864,6 +864,7 @@ func TestOIDCSetup(t *testing.T) {
 			backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				requestDump, _ := httputil.DumpRequest(r, false)
 				assert.Contains(t, string(requestDump), tc.expectRequest, "expected request not fulfilled")
+				assert.NotContains(t, string(requestDump), oauthOidcCookieName, oauthOidcCookieName+" cookie should be dropped")
```
Please extend or add a test to verify that _other_ cookies are passed es expected

AlexanderYastrebov commented:
@AndreasSko FYI, I am trying what Copilot review is capable of so I've requested a review from it.

AndreasSko commented:
@AlexanderYastrebov thank you for the review! I adjusted the code 🙂
In the process I noticed a bug in the `insecureCookieJar`, which I fixed as well (1889ea326d9c65477d90ff354b1e922b5389c1b2).

AlexanderYastrebov commented on filters/auth/oidc_test.go:
```diff
@@ -950,6 +974,10 @@ func TestOIDCSetup(t *testing.T) {
 				Jar:     newInsecureCookieJar(),
 			}
 
+			for _, c := range tc.cookies {
+				client.Jar.SetCookies(reqURL, []*http.Cookie{c})
+			}
```
Shouldn't it be enough to
```go
client.Jar.SetCookies(tc.cookies)
```
?

AndreasSko replied:
Oh yeah, of course 🤦‍♂️ Good catch! 😄

AlexanderYastrebov commented:
This feels somehow related to https://github.com/zalando/skipper/pull/1965

AndreasSko commented:
> This feels somehow related to #1965

Yeah, true. Would you like me to adjust anything in the test?

AlexanderYastrebov commented:
:+1:

AlexanderYastrebov commented:
Thank you for improving Skipper.

AlexanderYastrebov commented:
@AndreasSko Could you please squash and rebase on top of master and make proper commit message. PR description and title should match those of commit message - the reason is that we release each merge to master so commit message becomes release notes.

AndreasSko commented:
Sure! Squashed and adjusted the commit message to match the PR description 🙂

AlexanderYastrebov commented:
:+1:

AndreasSko commented:
@AlexanderYastrebov @MustafaSaber any update on this one? We are very eager to use it soon 😁😊

MustafaSaber commented:
👍
